unblob/handlers/executable/elf.py

```diff
@@ -26,6 +28,27 @@
 KERNEL_MODULE_SIGNATURE_FOOTER = b"~Module signature appended~\n"
 
 
+@attr.define(repr=False)
+class ElfChunk(ValidChunk):
+    def extract(self, inpath: Path, outdir: Path):
+        # ELF file extraction is special in that in the general case no new files are extracted, thus
+        # when we want to clean up all carves to save place, carved ELF files would be deleted as well,
+        # however we want to keep carved out ELF files, as they are the interesting stuff!
+        if self.is_whole_file:
+            # whole file chunks are not carved, so we can do normal extraction
+            super().extract(inpath, outdir)
+        else:
+            # make a copy, and let the carved chunk be deleted
+            outdir.mkdir(parents=True, exist_ok=False)
+            shutil.copy2(inpath, outdir)
+            # more work will be done, when outdir is picked up by processing,
+            # and the ELF file is processed as a whole file.
+            # As a performance side effect, ELF files will be searched for chunks twice.
+            # Even though the second chunk search one is short-circuited,
+            # because the ELF handler will recognize it as a whole file
+            # other handlers might burn some cycles on the file as well.
```

Comment:
Theoretically this is what happens with any plaintext structure like extfs or tar, the container is scanned (firs scan), files are extracted, then files are scanned (second scan). Usually optimized by streaming mode and skipping candidates within a valid chunk.

Reply:
Well, this is different here, as ELF file plays the role of extfs/tar in your example, yet the ELF file is scanned twice, once as a chunk, then as a whole file:

- ELF chunk is recognized by scanning a file
- chunk is extracted, and processed by ELF handler, which just copies it to the extraction directory
- extraction directory is added as sub-task
- eventually the copied ELF file is scanned again without knowing it is an ELF file
- the processing of whole file non-kernel ELF files is just a no-op, so nothing is done

This is different from how an extfs chunk is handled, as that chunk is extracted in one step.
Actually this difference is the whole reason behind this MR: while throwing away all chunks, we want to keep extracted ELF chunks for further analysis.